x-pack/plugins/transform/public/app/sections/transform_management/components/transform_list/transform_search_bar.tsx

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Dispatch, SetStateAction, FC, Fragment, useState } from 'react';
+import {
+  EuiBadge,
+  EuiSearchBar,
+  EuiSearchBarProps,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { TermClause, FieldClause, Value } from './common';
+import { TRANSFORM_STATE } from '../../../../../../common';
+import { TRANSFORM_MODE, TransformListRow } from '../../../../common';
+import { getTaskStateBadge } from './use_columns';
+
+const filters = [
+  {
+    type: 'field_value_selection' as const,
```

Comment:
nit: not sure if you need the `as const` here and for the ones below. Maybe we can set `const filters: SearchFilterConfig[]` and TS will know to cast those fields correctly?

Reply:
Updated to use `SearchFilterConfig` in 9c2d7d1935a8d069b9df63af19dade005a99e624